packages/react-i18n/src/i18n.ts

```diff
@@ -231,6 +254,47 @@ export class I18n {
     return this.formatNumber(amount, {as: 'currency', ...options});
   }
 
+  formatCurrencyExplicit(
+    amount: number,
+    options: Intl.NumberFormatOptions = {},
+  ): string {
+    const value = this.formatCurrencyShort(amount, options);
+    const isoCode = options.currency || this.defaultCurrency || '';
+    if (value.includes(isoCode)) {
+      return value;
+    }
+    return `${value} ${isoCode}`;
+  }
+
+  formatCurrencyShort(
+    amount: number,
+    options: NumberFormatOptions = {},
+  ): string {
+    const {locale} = this;
+    const sym = this.getShortCurrencySymbol(options.currency);
+    let adjustedPrecision = options.precision;
+    if (adjustedPrecision === undefined) {
+      const currency = options.currency || this.defaultCurrency || '';
+      if (currency in CURRENCY_DIGITS) {
+        adjustedPrecision = CURRENCY_DIGITS[currency];
+      } else {
+        adjustedPrecision = 2;
+      }
+    }
+    const formattedAmount = memoizedNumberFormatter(locale, {
+      style: 'decimal',
+      minimumFractionDigits: adjustedPrecision,
+      maximumFractionDigits: adjustedPrecision,
+      ...options,
+    }).format(amount);
+
+    if (sym.prefixed) {
```

Comment:
We could clean this one up a bit as well. 
```
return sym.prefix ?  sym.symbol + formattedAmount : formattedAmount + sym.symbol;
```

Reply:
Also, it looks like we joining values with symbols for formatting? Much safer to use template strings in JS instead of concatenating strings like this.
 
```
return sym.prefix ?  `${sym.symbol}  ${formattedAmount}` : `${formattedAmount} ${sym.symbol}`;
```

Reply:
Adjusting as suggested--thank you.